sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AbstractSqlParser.scala

```diff
@@ -91,6 +91,20 @@ abstract class AbstractSqlParser extends AbstractParser with ParserInterface {
     }
   }
 
+  /** Creates BatchBody for a given SQL string. */
+  override def parseBatch(sqlText: String): BatchBody = parse(sqlText) { parser =>
+    val ctx = parser.batchOrSingleStatement()
+    withErrorHandling(ctx, Some(sqlText)) {
+      astBuilder.visitBatchOrSingleStatement(ctx) match {
+        case batch: BatchBody => batch
+        case _ =>
+          // TODO: Figure out the error reporting - this is just copy paste of parsePlan logic.
```

Comment:
Not sure what to do about this?
We would be throwing exceptions from `SqlBatchLangErrors` for anything related to our parsing, interpreting, debugging, etc.
So maybe create/reuse some generic exception saying like "Unable to parse..." or "Internal error" type of exception?

Reply:
We synced offline and decided to leave "Unsupported SQL statement" exception here as in `parsePlan`. As already explained, all meaningful exceptions during parsing, interpreting, executing, etc. would be thrown elsewhere.

This also brings down the scope of this PR to exclude exception support - which will be added anyways, and if we want we can easily change this exception here in the future.